Modules/FV0/src/TriggersSwVsTcmCheck.cxx

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TriggersSwVsTcmCheck.cxx
+/// \author Dawid Skora dawid.mateusz.skora@cern.ch
+///
+
+#include "FV0/TriggersSwVsTcmCheck.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+// ROOT
+#include <TH1.h>
+#include <TH2.h>
+#include <TPaveText.h>
+#include <TLine.h>
+#include <TList.h>
+
+#include <DataFormatsQualityControl/FlagReasons.h>
+
+using namespace std;
+using namespace o2::quality_control;
+
+namespace o2::quality_control_modules::fv0
+{
+
+constexpr int kBinSwOnly = 1;
+constexpr int kBinTcmOnly = 2;
+
+void TriggersSwVsTcmCheck::configure() {
+    if (auto param = mCustomParameters.find("ccdbUrl"); param != mCustomParameters.end()) {
+      setCcdbUrl(param->second);
+      ILOG(Debug, Support) << "configure() : using deadChannelMap from CCDB, configured url = " << param->second << ENDM;
+    } else {
+      setCcdbUrl("o2-ccdb.internal");
+      ILOG(Debug, Support) << "configure() : using deadChannelMap from CCDB, default url = "
+                           << "o2-ccdb.internal" << ENDM;
+    }
+}
+
+Quality TriggersSwVsTcmCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  Quality result = Quality::Null;
+  int mNumErrors = 0;
+  for (auto& [moName, mo] : *moMap) {
+    (void)moName;
+    if (mo->getName() == "TriggersSoftwareVsTCM") {
+      auto* histogram = dynamic_cast<TH2F*>(mo->getObject());
+      result = Quality::Good;
+      int numberOfBinsX = histogram->GetNbinsX();
+      for (int i = 1; i <= numberOfBinsX; i++) {
+        if ((bool)(histogram->GetBinContent(i, kBinSwOnly)) == (bool)(histogram->GetBinContent(i, kBinTcmOnly))) {
```

Comment:
Won't this produce an error if `histogram->GetBinContent(i, kBinSwOnly) == histogram->GetBinContent(i, kBinTcmOnly) == 0`, which is a good situation? The triggers should always be in both SW and TCM, or neither

Reply:
Yes, you are right. Fixed in the latest commit.